### DISCUSSION TITLE
Custom Locales and Effective Locales

### DISCUSSION THREAD

**Discussion by davidmayr** [PaperMC/Paper#10570] [Ideas]
### Is your feature request related to a problem?

The Adventure Global Translator is an awesome feature and I love how it is automatically integrated and translates messages in various parts of paper (entity names, messages etc.)

However, when you have a custom language plugin one might want to sync the language of all other plugins and the global translator to the custom language you have stored in e.g. your database. 

### Describe the solution you'd like.

An API like I implemented in Velocity a few years back: https://github.com/PaperMC/Velocity/pull/537

This would allow all plugins to read the effective locale (a custom locale, the locale of the client or Locale.US) and provide the ability change it. getLocale() and locale() would be untouched.

This allows plugins to still read the players configured language but also allows plugins to change what language should be used.

### Describe alternatives you've considered.

Changing the language via NMS. But that's a bit hacky, and I'd like a proper solution

**Comment by Machine-Maker** [comment]
I think this is acceptable to add. Should be clear that this is only the locale being used to do the server-side translation, and won't change the locale the client actually is using.

**Comment by Laurenshup** [comment]
Hello, has this been implemented yet? As I am currently seeing myself finding this exact need.
